runtime/compiler/env/j9fieldsInfo.cpp

```diff
@@ -149,11 +149,12 @@ TR_VMFieldsInfo::TR_VMFieldsInfo(TR::Compilation * comp, J9Class *aClazz, int bu
       }
 
    //supers
-   J9Class **supClasses = aClazz->superclasses;
    int numSupClasses = J9CLASS_DEPTH(aClazz);
+   J9Class *supClass = aClazz;
    for (i=numSupClasses-1; i>=0; i--)
       {
-      J9Class *supClass = supClasses[i];
+      supClass = (J9Class*)comp->fej9()->getSuperClass((TR_OpaqueClassBlock*)supClass);
```

Comment:
At least at the use of this in ilgen, it's unfortunate to dig through all these superclasses just to [filter out fields inherited from them](https://github.com/eclipse/openj9/blob/35c44ead49a53f8c0ad1ca3e63c18711d3a7b26f/runtime/compiler/ilgen/Walker.cpp#L5515). If we find that this generates too many superclass records, we may want to put this use into a heuristic region. The information we're extracting is whether or not `methodClass` declares any of its own final (instance) fields, and while the use of that information isn't strictly heuristic, it is completely determined by the class chain of `methodClass`